PDFsharp/code/PdfSharp/PdfSharp.Drawing/XImage.cs

```diff
@@ -194,6 +194,21 @@ public static XImage FromFile(string path)
       return new XImage(path);
     }
 
+    /// <summary>
+    /// Creates an image from the specified stream.<br/>
+    /// Silverlight supports PNG and JPEG only.
+    /// </summary>
+    /// <param name="stream">The stream containing a BMP, PNG, GIF, JPEG, TIFF, or PDF file.</param>
+    public static XImage FromStream(Stream stream)
+    {
+	    if (stream == null)
+		    throw new ArgumentNullException("stream");
+
+	    if (PdfReader.TestPdfFile(stream) > 0)
+		    return new XPdfForm(stream);

```

Comment:
The return type for the method is different, do XPdfForm return type support ?

Reply:
XImage is parent class of XPdfForm